### PR TITLE
Enable C2 HIL

### PR DESCRIPTION
**Pull request by SergioGasquez**
## Thank you for your contribution!

We appreciate the time and effort you've put into this pull request.
To help us review it efficiently, please ensure you've gone through the following checklist:

### Submission Checklist 📝
- [x] I have updated existing examples or added new ones (if applicable).
- [x] I have used `cargo xtask fmt-packages` command to ensure that all changed code is formatted correctly.
- [x] My changes were added to the [`CHANGELOG.md`](https://github.com/esp-rs/esp-hal/blob/main/esp-hal/CHANGELOG.md) in the **_proper_** section.
- [x] My changes are in accordance to the [esp-rs API guidelines](https://github.com/esp-rs/esp-hal/blob/main/API-GUIDELINES.md)

#### Extra:
- [x] I have read the [CONTRIBUTING.md guide](https://github.com/esp-rs/esp-hal/blob/main/CONTRIBUTING.md) and followed its instructions.

#### Description
- Enables C2 HIL test suite
- Changes the wiring of all the current tests/targets, instead of connecting `GPIO2` and `GPIO4`, we should now use `GPIO2` and `GPIO3
  - This means that we should sync whenever we update the lab targets and when we merge this PR.
- Test that rely on `delay`s are failing, added the error to https://github.com/esp-rs/esp-hal/issues/1524 and disabled them for C2

> [!IMPORTANT]  
> Once merged we need to add the `HIL Test | esp32c2` test to the required checks

#### Testing
Locally run the test suite, modifying the connected pins, without errors for C2, C3, C6, H2, S2 and S3.

### CODE REVIEW

**Review by bjoernQ — APPROVED** [review]
LGTM

**Comment by SergioGasquez** [comment]
CI failing due to #1682, once #1681 is merged, a rebase should fix the errors.

**Comment by MabezDev** [comment]
What's the order of operations here? I guess we need to physically move the wires before we can merge this?

**Comment by SergioGasquez** [comment]
> What's the order of operations here? I guess we need to physically move the wires before we can merge this?

Yes, we would need to update the wiring and merge this asap after the wires are updated (otherwise HIL workflows will fail)

**Review by JurajSadel — APPROVED** [review]
LGTM, thanks! I can do the wiring changes.

**Comment by MabezDev** [comment]
@JurajSadel please go ahead and change the wires, then merge this once that's complete. Thank you :smiling_face_with_three_hearts:

**Comment by SergioGasquez** [comment]
Tomas just updated the wiring! Ill proceed with the merging

**Pull request merged**
